gtk4/src/subclass/editable.rs

```diff
@@ -0,0 +1,154 @@
+        editable_iface.get_selection_bounds = Some(editable_get_selection_bounds::<T>);
+        editable_iface.set_selection_bounds = Some(editable_set_selection_bounds::<T>);
+    }
+}
+
+unsafe extern "C" fn editable_insert_text<T: EditableImpl>(
+    editable: *mut gtk_sys::GtkEditable,
+    text: *const c_char,
+    length: c_int,
+    position: *mut c_int,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.insert_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        &GString::from_glib_borrow(text),
+        length,
+        &mut *position,
+    )
+}
+
+unsafe extern "C" fn editable_delete_text<T: EditableImpl>(
+    editable: *mut gtk_sys::GtkEditable,
+    start_position: c_int,
+    end_position: c_int,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.delete_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        start_position,
+        end_position,
+    )
+}
+
+unsafe extern "C" fn editable_changed<T: EditableImpl>(editable: *mut gtk_sys::GtkEditable) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.changed(from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref())
+}
+
+unsafe extern "C" fn editable_get_text<T: EditableImpl>(
+    editable: *mut gtk_sys::GtkEditable,
+) -> *const c_char {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.get_text(from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref())
+        .to_glib_full()
+}
+
+unsafe extern "C" fn editable_do_insert_text<T: EditableImpl>(
+    editable: *mut gtk_sys::GtkEditable,
+    text: *const c_char,
+    length: i32,
+    position: *mut i32,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.do_insert_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        &GString::from_glib_borrow(text),
+        length,
+        &mut *position,
+    )
+}
+
+unsafe extern "C" fn editable_do_delete_text<T: EditableImpl>(
+    editable: *mut gtk_sys::GtkEditable,
+    start_position: i32,
+    end_position: i32,
+) {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.do_delete_text(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        start_position,
+        end_position,
+    )
+}
+
+unsafe extern "C" fn editable_get_selection_bounds<T: EditableImpl>(
+    editable: *mut gtk_sys::GtkEditable,
+    start_position: *mut i32,
+    end_position: *mut i32,
+) -> glib_sys::gboolean {
+    let instance = &*(editable as *mut T::Instance);
+    let imp = instance.get_impl();
+
+    imp.get_selection_bounds(
+        from_glib_borrow::<_, Editable>(editable).unsafe_cast_ref(),
+        &mut *start_position,
+        &mut *end_position,
```

Comment:
Either or both of the pointers can be `NULL`. You need to check before assigning the results to them

Reply:
Should be ok now